packages/jest-cli/src/reporters/IstanbulTestReporter.js

```diff
@@ -37,6 +40,44 @@ class IstanbulTestReporter extends DefaultTestReporter {
         reporter.addAll(config.coverageReporters);
         reporter.write(collector, true, () => {});
       } catch (e) {}
+      if (config.coverageThreshold) {
+        const rawCoverage = collector.getFinalCoverage();
+        const globalResults = istanbul.utils.summarizeCoverage(rawCoverage);
+        
+        function check(name, thresholds, actuals) {
+          return [
+            'statements',
+            'branches',
+            'lines',
+            'functions',
+          ].reduce((errors, key) => {
+            const actual = actuals[key].pct,
+                actualUncovered = actuals[key].total - actuals[key].covered,
+                threshold = thresholds[key];
+
+            if (threshold < 0) {
+              if (threshold * -1 < actualUncovered) {
+                errors.push(
+                  `Uncovered count for ${key} (${actualUncovered})` +
+                  `exceeds ${name} threshold (${-1 * threshold})`
+                );
+              }
+            } else if (actual < threshold) {
+              errors.push(
+                `Coverage for ${key} (${actual}` +
+                `%) does not meet ${name} threshold (${threshold}%)`
+              );
+            }
+            return errors;
+          }, []);
+        }
+        const errors = check('global', config.coverageThreshold.global, globalResults);
+        
+        if (errors.length > 0) {
+          this.log(`${FAIL_COLOR(errors.join('\n'))}`);
+          aggregatedResults.success = false;
```

Comment:
I don't believe this will work. In TestRunner.js we define `success` as follows:

```js
aggregatedResults.success =
  aggregatedResults.numFailedTests === 0 &&
  aggregatedResults.numRuntimeErrorTestSuites === 0;
```

so it will be overwritten after the full run succeeds. We need to find another way to report back the failure. There is an `onRunComplete` on every reporter which can mutate `aggregatedResults`.